qabelbox/src/main/java/de/qabel/desktop/repository/sqlite/SqliteContactRepository.java

```diff
@@ -233,97 +234,115 @@ public Contact findByKeyId(String keyId) throws EntityNotFoundExcepion {
     }
 
     @Override
-    public List<Identity> findContactIdentities(String contactKey) throws PersistenceException, EntityNotFoundExcepion {
+    public Pair<Contact, List<Identity>> findContactWithIdentities(String key) throws PersistenceException, EntityNotFoundExcepion {
+        Contact contact = findByKeyId(key);
+
+        Identities identities = identityRepository.findAll();
+        Map<Integer, List<String>> associatedIdentities = findContactIdentityKeys(
+                Collections.singletonList(contact.getId()));
+        List<Identity> contactIdentities;
+        if (associatedIdentities.containsKey(contact.getId())) {
+            contactIdentities = new ArrayList<>(associatedIdentities.get(contact.getId()).size());
```

Comment:
I think initializing the ArrayList is not worth the hassle of retrieving the size in the first place. Afaik the size given to an ArrayList is a minimum size anyway, so the small value here is probably ignored anyway.